### PR TITLE
use Posix.read instead of Glibc.read for timerfd 

### PR DESCRIPTION
**Pull request by shekhar-rajak**
### Motivation:

To read the bytes from `timerfd`, we use Glibc.read directly but we should use `try! Posix.read`

### Modifications:

Using `Posix` instead of `Glibc` for read operation with `try!` to check for any error.

### Result:

Fixes https://github.com/apple/swift-nio/issues/1341

### CODE REVIEW

File: Sources/NIO/Selector.swift [hunk 1]
```diff
@@ -549,7 +549,7 @@ internal class Selector<R: Registration> {
                 // Consume event
                 var val: UInt64 = 0
                 // We are not interested in the result
-                _ = Glibc.read(self.timerFD, &val, MemoryLayout.size(ofValue: val))
+                _ = try! Posix.read(self.timerFD, &val, MemoryLayout.size(ofValue: val))
```
Comment (glbrntt) [inline Sources/NIO/Selector.swift]: Looks like you're missing some labels here: https://github.com/apple/swift-nio/blob/546a84c4e11b8100b962c55afc77f7c3f94d1d2c/Sources/NIO/System.swift#L396

Comment (shekhar-rajak) [reply Sources/NIO/Selector.swift]: Thanks for checking. I am figuring out the values.

**Comment by weissi** [comment]
@swift-nio-bot test this please

**Comment by shekhar-rajak** [comment]
I didn't understand the  CI error. Please help me to understand it.

**Comment by weissi** [comment]
ignore it, our CI is underpowered is what that means :(

**Comment by weissi** [comment]
@swift-nio-bot test this please

**Comment by shekhar-rajak** [comment]
Thanks! :)

**Pull request merged**